react-native/react/login2/register/set-public-name/index.js

```diff
@@ -15,11 +15,14 @@ export default class SetPublicName extends Component {
     }
   }
 
-  submit () {
-    this.props.submit(this.state.deviceName)
+  onSubmit () {
+    this.props.onSubmit(this.state.deviceName)
   }
 
   render () {
+    const dupeName = this.props.existingDevices && this.props.existingDevices.indexOf(this.state.deviceName) !== -1
```

Comment:
I think that's the right solution here, but I'm okay with that happening in another PR as long as we don't forget about it